docs/src/modules/components/demos/data-grid/Inventory/InventoryDashboard.tsx

```diff
@@ -0,0 +1,257 @@
+  },
+  {
+    field: 'rating',
+    headerName: 'Rating',
+    aggregable: false,
+    width: 150,
+    flex: 1,
+    renderCell: (params: GridRenderCellParams<Product, number>) =>
+      params.id === GRID_AGGREGATION_ROOT_FOOTER_ROW_ID ? (
+        <Typography variant="body2">Sales:</Typography>
+      ) : (
+        <Rating value={params.value} precision={0.1} readOnly size="small" />
+      ),
+  },
+  {
+    field: 'sales',
+    type: 'number',
+    headerName: 'Sales',
+    width: 100,
+    flex: 1,
+    availableAggregationFunctions: ['sum'],
+  },
+];
+
+function InventoryDashboard() {
+  const [searchQuery, setSearchQuery] = React.useState('');
+  const [statusFilter, setStatusFilter] = React.useState('all');
+  const apiRef = useGridApiRef();
+
+  const onRowClick = React.useCallback<GridEventListener<'rowClick'>>(
+    (params) => {
+      apiRef.current?.toggleDetailPanel(params.id);
+    },
+    [apiRef],
+  );
+
+  const filteredProducts = React.useMemo(() => {
+    return products.filter((product) => {
+      const matchesSearch =
+        product.product.toLowerCase().includes(searchQuery.toLowerCase()) ||
+        product.sku.toLowerCase().includes(searchQuery.toLowerCase());
+      const matchesStatus = statusFilter === 'all' || product.status === statusFilter;
+      return matchesSearch && matchesStatus;
+    });
+  }, [searchQuery, statusFilter]);
+
+  const profitAggregation: GridAggregationFunction<string, string | null> = {
+    apply: (params) => {
+      const totalProfit = products.reduce(
+        (acc, product) => acc + product.sales * (product.price - product.cost),
+        0,
+      );
+      return totalProfit != null ? `${totalProfit}` : null;
+    },
+  };
+
+  return (
+    <DemoContainer theme={inventoryTheme}>
+      <ThemeProvider theme={inventoryTheme}>
+        <Box
+          sx={{
+            display: 'flex',
+            flexDirection: 'column',
+            gap: 2,
+            height: '100%',
+            width: '100%',
+          }}
+        >
+          <Paper
+            elevation={0}
+            sx={{
+              p: 2,
+              border: '1px solid',
+              borderColor: 'divider',
+              borderRadius: '16px',
+              height: '100%',
+              display: 'flex',
+              flexDirection: 'column',
+            }}
+          >
+            <InventoryToolbar
+              searchQuery={searchQuery}
+              onSearchChange={setSearchQuery}
+              statusFilter={statusFilter}
+              onStatusChange={setStatusFilter}
+            />
+            <Box sx={{ flex: 1, display: 'flex', flexDirection: 'column', minHeight: 0 }}>
+              <DataGridPremium<Product>
+                apiRef={apiRef}
+                rows={filteredProducts}
+                columns={columns}
+                aggregationFunctions={{ ...GRID_AGGREGATION_FUNCTIONS, profit: profitAggregation }}
```

Comment:
What was the purpose for custom aggregation function in this demo?
Aggregation only works with row grouping (or tree data) enabled.
If row grouping is not going to be used in this demo, the aggregation-related code can be removed here.

Reply:
Heh, yeah! I'm just trying to add as many features as I can for now for when I work on a written/video walkthrough.